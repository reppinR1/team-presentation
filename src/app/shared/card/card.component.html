<ng-content select="[image]"></ng-content>
<ng-content select="[title]"></ng-content>
<ng-content select="[secondary-title]"></ng-content>
<div class="content">
  <ng-content select="[content]"></ng-content>
</div>
