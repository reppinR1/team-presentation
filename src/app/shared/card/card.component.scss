:host {
  display: block;
}

.image-wrapper {
  margin-bottom: 24px;
}
