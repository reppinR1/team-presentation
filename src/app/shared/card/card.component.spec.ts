import { createHostFactory, SpectatorHost } from '@ngneat/spectator';
import { CardComponent } from './card.component';

describe('CardComponent', () => {
  let spectator: SpectatorHost<CardComponent>;
  const createHost = createHostFactory({
    component: CardComponent,
    shallow: true,
  });

  it('should project image into image wrapper', () => {
    spectator = createHost(`
    <app-card>
      <picture image></picture>
    </app-card>
    `);
    expect(spectator.fixture).toMatchSnapshot();
  });
  it('should project content into content section', () => {
    spectator = createHost(`
    <app-card>
      <section content></section>
    </app-card>
    `);
    expect(spectator.fixture).toMatchSnapshot();
  });
});
