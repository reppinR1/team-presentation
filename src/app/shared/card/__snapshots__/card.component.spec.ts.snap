// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardComponent should project content into content section 1`] = `
<ng-component
  __ngContext__={[Function LRootView]}
>
  <app-card>
    <div
      class="image-wrapper"
    />
    <section
      content=""
    />
  </app-card>
</ng-component>
`;

exports[`CardComponent should project image into image wrapper 1`] = `
<ng-component
  __ngContext__={[Function LRootView]}
>
  <app-card>
    <div
      class="image-wrapper"
    >
      <picture
        image=""
      />
    </div>
  </app-card>
</ng-component>
`;
